crates/router/src/connector/nexinets/transformers.rs

```diff
@@ -525,13 +586,88 @@ fn get_payment_details_and_product(
     }
 }
 
-fn get_card_details(req_card: &api_models::payments::Card) -> NexinetsPaymentDetails {
-    NexinetsPaymentDetails::Card(Box::new(NexiCardDetails {
+fn get_card_data(
+    item: &types::PaymentsAuthorizeRouterData,
+    card: &api_models::payments::Card,
+) -> Result<NexinetsPaymentDetails, errors::ConnectorError> {
+    if item.request.is_mandate_payment() {
+        let cof_contract = Some(CofContract {
+            recurring_type: RecurringType::Unscheduled,
+        });
+        let card_data = if item.request.off_session.is_some() {
+            CardDataDetails::PaymentInstrument(Box::new(get_payment_instrument(item)))
+        } else {
+            CardDataDetails::CardDetails(Box::new(get_card_details(card)))
+        };
+        Ok(NexinetsPaymentDetails::Card(Box::new(NexiCardDetails {
+            card_data,
+            cof_contract,
+        })))
+    } else {
+        Ok(NexinetsPaymentDetails::Card(Box::new(NexiCardDetails {
+            card_data: CardDataDetails::CardDetails(Box::new(get_card_details(card))),
+            cof_contract: None,
+        })))
+    }
+}
+
+fn get_payment_instrument(item: &types::PaymentsAuthorizeRouterData) -> PaymentInstrument {
+    let payment_instrument_id = item
+        .request
+        .mandate_id
+        .as_ref()
+        .and_then(|mandate_ids| mandate_ids.connector_mandate_id.clone());
+    PaymentInstrument {
+        payment_instrument_id,
+    }
+}
+
+fn get_applepay_details(
+    wallet_data: &api_models::payments::WalletData,
+    applepay_data: &api_models::payments::ApplePayWalletData,
+) -> CustomResult<ApplePayDetails, errors::ConnectorError> {
+    let payment_data = wallet_data.get_wallet_token_as_json()?;
+    Ok(ApplePayDetails {
```

Comment:
is this tested?

Reply:
This is not tested . Although i have compared the request format after building request .